tensorflow_datasets/core/features/image_feature.py

```diff
@@ -189,14 +209,36 @@ def to_json_content(self) -> Json:
     }
 
 
-def create_thumbnail(ex):
+def create_thumbnail(ex, use_colormap: bool):
   """Creates the image from the np.array input."""
   PIL_Image = lazy_imports_lib.lazy_imports.PIL_Image  # pylint: disable=invalid-name
 
   _, _, c = ex.shape
   postprocess = _postprocess_noop
   if c == 1:
-    ex = ex.squeeze(axis=-1)
+    if use_colormap:
+      matplotlib = lazy_imports_lib.lazy_imports.matplotlib  # pylint: disable=invalid-name
+      mcolors = matplotlib.colors
+
+      ex = ex.squeeze(axis=-1)
+      unique_pixels = np.unique(ex)
+      blank_image = np.ones(shape=(*ex.shape[:2], 3), dtype=np.uint8)
+
+      if len(unique_pixels) <= 7:
```

Comment:
If in image1 there is 5 values, but in image 2 there is 10, then colors won't match between images.

We should make sure colors are consistents so that `32` is always matched with the same color, irrespectively of the image. And independently if `np.unique(ex) == [0, 32]`, `np.unique(ex) == [0, 2, 3, 4 32]`,...